From d631c1efdaa7b39fe3fadf906328951d58807caa Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Fri, 5 Aug 2022 16:00:56 +0800
Subject: [PATCH] [riscv64]  disable fp multiply and accumulate instructions

Some wasm interpreter tests are failing since instructions generated
by gcc such as *multiply and and* (fmadds) create intermediate
results bigger than 8 bytes which doesn't match other architectures,
hence the resulting output differs.

Port commit 13314a207e2b22a10c72bfa83f985c7ff144c775

co-authors: Jun Yuan Tan <junyuan.tan@starfivetech.com>

Change-Id: I18c0b659f30df84bb30daa176368a7e81b51063e
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3811139
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Clemens Backes <clemensb@chromium.org>
Cr-Commit-Position: refs/heads/main@{#82240}
---
 BUILD.gn | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index 0c1a72dfd2a..3a9407cdc16 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1244,6 +1244,9 @@ config("toolchain") {
     defines += [ "V8_TARGET_ARCH_RISCV64" ]
     defines += [ "__riscv_xlen=64" ]
     defines += [ "CAN_USE_FPU_INSTRUCTIONS" ]
+    if (!is_clang) {
+      cflags += [ "-ffp-contract=off" ]
+    }
     if (target_is_simulator) {
       defines += [ "CAN_USE_RVV_INSTRUCTIONS" ]
     }
-- 
2.35.1

